tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/object/test_bir_positive.bal

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import bir/objs;
+
+objs:Qux quxMod = isolated service object {
+    public isolated function execute() returns anydata|error {
+         return "qux";
+    }
+};
+
+function f1(objs:Foo foo) {
+    assertTrue(foo is objs:Foo);
```

Comment:
I would also add some compile-time positive tests like

```ballerina
T _ = t; // where `t` will be of a different type
```

Multiple combinations
- source and target from a imported module
- source from current module, target from imported module
- source from imported module, target from current module

Reply:
These combinations can also be covered in runtime positive (`is`) tests.

Reply:
Added in [#050dc4](https://github.com/ballerina-platform/ballerina-lang/pull/40901/commits/050dc4b510554b9df7e59929bf5265976524497b).